.github/workflows/publish.yml

```diff
@@ -0,0 +1,47 @@
+name: Build and Push Docker Images to GitHub Packages
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        target:
+            - x86_64-unknown-linux-musl
+            - aarch64-unknown-linux-musl
+            - armv7-unknown-linux-musleabihf
+            - i686-unknown-linux-musl
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: all
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and Push ${{ matrix.target }}
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          push: true
+          tags: ghcr.io/${{ github.repository_owner }}/rust-musl-cross:${{ matrix.target }}
```

Comment:
I don't bother creating a separate tag with `unknown-linux` omitted. I guess they do this upstream, so it might be worth it here. I don't really see what the purpose is though.